src/handle-data.ts

```diff
@@ -85,6 +85,16 @@ export function getTotalPosition(layout: TableLayout) {
   return 'noTotals';
 }
 
+export function getProperties(reference: string, data: any, defaultValue?: undefined) {
+  const steps: string[] = reference.split('.');
+  if (typeof data === 'undefined') return defaultValue;
+  let value = data;
+  for (let i = 0; i < steps.length; ++i) {
+    value[steps[i]] == null ? () => defaultValue : (value = value[steps[i]]);
+  }
+  return value;
+}
+
```

Comment:
This looks like optional chaining to me, especially since the current use-case does not pass a default value. Have you tried that? Also not sure this should be in handle-data

Reply:
Yes it is true, the default value doesn't paly any role here...
It is actually what is used in sense-client, I was not sure about the conditions, I just kept it because I thought we might reuse the function somewhere else. I'll remove it.

Reply:
BTW, data type is `HypercubeHandler` I don't think we have defined such a type anywhere.

Reply:
hmm no, I don't recognize it. But it is only in the ext file right?

Reply:
Yes

Reply:
Then don't mind it. If we want to refactor and convert ext file that becomes relevant